src/ui/src/components/blueprints/base/BlueprintToolbar.vue

```diff
@@ -92,11 +99,12 @@ async function runBlueprint(componentId?: string) {
 			class="BlueprintToolbar__runBlueprint"
 			data-automation-action="run-blueprint"
 			variant="special"
-			:disabled="isRunning"
-			@click="runBlueprint()"
+			@click="isRunning ? stopBlueprintRun() : runBlueprint()"
 		>
-			<i class="material-symbols-outlined">play_arrow</i>
-			{{ isRunning ? "Running..." : "Run blueprint" }}
+			<i class="material-symbols-outlined">{{
+				isRunning ? "stop" : "play_arrow"
+			}}</i>
+			{{ isRunning ? "Stop run" : "Run blueprint" }}
```

Comment:
Don't you think we should use "Cancel" for the button ? 🤔 

```suggestion
			{{ isRunning ? "Cancel" : "Run blueprint" }}
```

Reply:
I found "Stop run" somewhere on designs and to be honest I don't have any opinion here 🤷